test/TestServerProjectsLowLevel/dpg-customization/Generated/DPGClient.cs

```diff
@@ -0,0 +1,415 @@
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace dpg_customization_LowLevel
+{
+    /// <summary> The DPG service client. </summary>
+    public partial class DPGClient
+    {
+        private const string AuthorizationHeader = "Fake-Subscription-Key";
+        private readonly AzureKeyCredential _keyCredential;
+        private readonly HttpPipeline _pipeline;
+        private readonly Uri _endpoint;
+
+        /// <summary> The ClientDiagnostics is used to provide tracing support for the client library. </summary>
+        internal ClientDiagnostics ClientDiagnostics { get; }
+
+        /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+        public virtual HttpPipeline Pipeline => _pipeline;
+
+        /// <summary> Initializes a new instance of DPGClient for mocking. </summary>
+        protected DPGClient()
+        {
+        }
+
+        /// <summary> Initializes a new instance of DPGClient. </summary>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="credential"/> is null. </exception>
+        public DPGClient(AzureKeyCredential credential, Uri endpoint = null, DPGClientOptions options = null)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
+            endpoint ??= new Uri("http://localhost:3000");
+            options ??= new DPGClientOptions();
+
+            ClientDiagnostics = new ClientDiagnostics(options);
+            _keyCredential = credential;
+            _pipeline = HttpPipelineBuilder.Build(options, Array.Empty<HttpPipelinePolicy>(), new HttpPipelinePolicy[] { new AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) }, new ResponseClassifier());
+            _endpoint = endpoint;
+        }
+
+        /// <summary> Get models that you will either return to end users as a raw body, or with a model added during grow up. </summary>
+        /// <param name="mode"> The mode with which you&apos;ll be handling your returned body. &apos;raw&apos; for just dealing with the raw body, and &apos;model&apos; if you are going to convert the raw body to a customized body before returning to users. </param>
+        /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="mode"/> is null. </exception>
+        /// <remarks>
+        /// Schema for <c>Response Body</c>:
+        /// <code>{
+        ///   received: &quot;raw&quot; | &quot;model&quot;
+        /// }
+        /// </code>
+        /// 
+        /// </remarks>
+        public virtual async Task<Response> GetModelAsync(string mode, RequestContext context = null)
```

Comment:
Where is the grow-up method that returns `Task<Response<SomeModel>`?

Per the scenario: **"Improve a GET method that return raw JSON to return a model"**

Reply:
I see we have used a cast from Response to the model type (approach #2) rather than adding the grow-up method (approach #4).  I will check with Krzysztof on what we want to validate for this sprint.

Reply:
I would agree that we need to validate for this sprint. Otherwise I couldn't find this issue https://github.com/Azure/autorest.csharp/issues/2014.

Reply:
We have decided to go with approach 4, adding models into the main assembly and adding convenience methods to clients.

I answered the question in your issue - let me know if you run into more issues with this.